src/dlls/mscoree/mscoree.cpp

```diff
@@ -907,9 +907,23 @@ STDAPI GetCORSystemDirectoryInternal(__out_ecount_part_opt(cchBuffer, *pdwLength
     if (pBuffer == NULL)
         IfFailGo(E_POINTER);
 
+#ifdef CROSSGEN_COMPILE
```

Comment:
The PAL_GetPALDirectory gets it from the pal_module and that one is initialized only when the PAL is initialized using the PAL_InitializeCoreCLR. And it is a misnomer now, since it returns libcoreclr path (there is no separate PAL module) as John has mentioned.